### PR TITLE
fix(settings): fix list spacing #5464

### PR DESCRIPTION
**Pull request by petermakowski**
## Done
- Fix values spacing when adding a package repository
- update the parsing logic to correctly split the values and remove any leading or trailing spaces

This fixes an issue where the MAAS UI incorrectly parses comma-separated distributions and components when adding a new package repository.

Fixes: #5464

## QA

1. Go to Settings->Package Repos->Add Repository
2. For both the Distributions and Components sections, enter any values separated with commas (e.g. `focal, jammy, noble` or `focal,jammy,noble`)
3. Click Save Repository
4. Inspecting the WS request sent to MAAS and ensure the list  is `["focal", "jammy", "noble"]`

### CODE REVIEW

File: src/app/utils/parseCommaSeparatedValues.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const parseCommaSeparatedValues = (input: string): string[] => {
+  return input
+    .split(",")
+    .map((s) => s.trim())
+    .filter(Boolean);
```
Comment (ndv99) [inline src/app/utils/parseCommaSeparatedValues.ts:5]: What does this line do?

**Review by ndv99 — APPROVED** [review]
LGTM, just one question below

File: src/app/utils/parseCommaSeparatedValues.ts (hunk 1 above)
Comment (petermakowski) [reply src/app/utils/parseCommaSeparatedValues.ts:5]: `Boolean` converts any given value to a boolean type. It returns true for truthy values and false for falsy values. Here it's used as an argument in `.filter()`effectively filtering out all falsy values from the array.

Comment (petermakowski) [reply src/app/utils/parseCommaSeparatedValues.ts:5]: Examples below might help if still unclear.

```tsx
"first, , third".split(",")
    .map((s) => s.trim())
```
`['first', '', 'third']`

---

```tsx
"first, , third".split(",")
    .map((s) => s.trim())
    .filter(Boolean);
```
`['first', 'third']`

**Pull request merged**